src/template.js

```diff
@@ -35,6 +38,30 @@ function logger(message, event) {
   log('[GTM-DEBUG] Search Insights > ' + message, event || '');
 }
 
+function clone(obj) {
+  const keys = Object.keys(obj);
+  const newObj = {};
+  keys.forEach((key) => {
+    newObj[key] = obj[key];
+  });
+  return newObj;
+}
+
+function chunkPayload(payload, keys, limit) {
+  // This assumes the values of `keys` have the same length.
```

Comment:
I would do the check and bail out early if it fails instead of assuming the behaviour.

```suggestion
  // check if the values in `payload` for each of `keys` have the same length.
  const sameNumberOfValues = keys.map(k => payload[k].length).every(n => n === payload[keys[0]].length)
  if (!sameNumberOfValues) {
    // chunking behaviour is unsafe due to unequal length arrays to chunk.
    // bail out early.
    return [payload]
  }
```

Reply:
I have a question. If there's 30 object ids and 25 positions, will the api respond as "number mismatch" or "exceeding limit"?

If former, they will learn it from the events debugger, and fix it. But if latter, there's another hidden problem waiting for them, and could be surprising because they've never dealt with chunking (because the connector did it for them)

Reply:
Just tested it and it will warn about the event number mismatch first:
```sh
curl -X POST \
https://insights.algolia.io/1/events \
-H 'x-algolia-api-key:${API_KEY}' --referer 'http://example.com/debug' \
-H 'x-algolia-application-id: ${APP_ID}' -H 'x-algolia-agent: insights-js (2.1.0); insights-gtm (1.2.1)' \
-H "Content-Type: application/json" \
-d '{
    "events": [
        {
          "eventType": "click",
          "eventName": "Product Clicked",
          "index": "products",
          "userToken": "user-123456",
          "timestamp": 1638780188289,
          "objectIDs": ["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23"],
          "queryID": "43b15df305339e827f0ac0bdc5ebcaa7",
          "positions": [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21]
        }
      ]
    }'

{"status":422,"message":"Event of type click should have the same number of ObjectIDs and Positions"}
```